lib/sippy_cup/scenario.rb

```diff
@@ -156,7 +156,9 @@ def invite(opts = {})
       # FIXME: The DTMF mapping (101) is hard-coded. It would be better if we could
       # get this from the DTMF payload generator
       from_addr = "#{@from_user}@#{@adv_ip}:[local_port]"
-      msg = <<-MSG
+      opt_hdrs = (opts.has_key?(:auth_keyword) ? opts[:auth_keyword] : '')
+      + (opts.has_key?(:headers) ? "\n#{opts[:headers]}" : '' ) 
+      msg = <<-HDR + (opt_hdrs.nil?||opt_hdrs.empty? ? '' : (opt_hdrs + "\n")) + "\n" + <<-SDP
```

Comment:
This is a bit confusing because you are defining a nested HEREDOC block but not closing them in a nested fashion, i.e.
```
<<-HDR 
# some stuff
<<-SDP
# some stuff
HDR
# some stuff
SDP
```
You should be able to do this in a single heredoc block and use string interpolation synatx `#{}` to inject the ruby block you want